ProjFS.Mac/PrjFSLib/PrjFSUser.cpp

```diff
@@ -122,3 +136,98 @@ bool PrjFSService_DataQueueInit(
     return false;
 }
 
+IOReturn DataQueue_Dequeue(IODataQueueMemory* dataQueue, void* data, uint32_t* dataSize)
+{
+    if (nullptr == ioDataQueueDequeueFunction)
+    {
+        InitDataQueueFunctions();
+    }
+    return ioDataQueueDequeueFunction(dataQueue, data, dataSize);
+}
+
+IODataQueueEntry* DataQueue_Peek(IODataQueueMemory* dataQueue)
+{
+    if (nullptr == ioDataQueuePeekFunction)
+    {
+        InitDataQueueFunctions();
+    }
+    return ioDataQueuePeekFunction(dataQueue);
+}
+
+
+static bool GetDarwinVersion(DarwinVersion& outVersion)
+{
+    utsname unameInfo = {};
+    if (0 != uname(&unameInfo))
+    {
```

Comment:
I'm unfortunately not aware of a nicer way to get the macOS version with just the C system libraries. (as opposed to Objective-C/Cocoa) There's also a "kern.osrelease" sysctl, but that returns the exact same version string as `uname()` that still needs to be parsed.

Reply:
Did you consider `sw_vers -productVersion`? That should give you the macOS version number, not Darwin. You'd still have to do some parsing though.

Reply:
That would require shelling out or reading the file `/System/Library/CoreServices/SystemVersion.plist` and subsequently parsing the string anyway as you say. I think a simple syscall is preferable in that regard. The darwin version tends to change on very minor revisions (security updates, supplementary updates, etc.) too, so in theory that would allow us to turn off the workaround if Apple changed their mind and supplied a fix with a supplementary update to 10.13.6.